experimental/dds/tree/src/SharedTree.ts

```diff
@@ -938,10 +938,10 @@ export class SharedTree extends SharedObject<ISharedTreeEvents> implements NodeI
 			} else if (type === SharedTreeOpType.Edit) {
 				if (op.version === WriteFormat.v0_1_1) {
 					// TODO: This cast can be removed on typescript 4.6
```

Comment:
Can remove this comment and the one below. I see you removed some others, so probably just forgot these ones.

Reply:
Interesting that some but not all of the as casts were build errors due to unneeded casts. I just fixed the build error ones, but looks like there were more. Fixing now.

Reply:
All cleaned up now, I think. I had to leave one of the casts and todos: I guess it needs TS 4.6 not just ts 4.5.